mobile-app/app/screens/WalletNavigator/screens/CreateWallet/PinConfirmation.tsx

```diff
@@ -47,12 +52,35 @@ export function PinConfirmation ({ route }: Props): JSX.Element {
       MnemonicEncrypted.toData(copy.words, copy.network, copy.pin)
         .then(async encrypted => {
           await MnemonicStorage.set(words, pin)
+          if (type === 'restore') {
+            await discoverWalletAddresses(encrypted)
+          }
           await setWallet(encrypted)
         })
         .catch(logger.error)
     }, 50) // allow UI render the spinner before async task
   }
 
+  async function discoverWalletAddresses (data: WalletPersistenceDataI<EncryptedProviderData>): Promise<void> {
```

Comment:
Don't create another instance of Wallet, after `setWallet(encrypted)` on [line 58](https://github.com/DeFiCh/wallet/pull/1274/files#diff-0c3a3beb288ddbce90bb62f5e179694e7431c6feed2bda726bef9f5430f75339R58), use that newly initialized wallet and use it for discovery.

Reply:
@fuxingloh , in set wallet  we set the length of the wallet, here after that we directly switch the component (check the below logic)
```
if (wallets.length === 0) {
    return <WalletNavigator />
  }

  return (
    <WalletNodeProvider data={wallets[0]}>
      <WalletContextProvider api={WalletAddressIndexPersistence}>
        <PrivacyLock />
        <BottomSheetModalProvider>
          <TransactionAuthorization />
          <AppNavigator />
        </BottomSheetModalProvider>
      </WalletContextProvider>
    </WalletNodeProvider>
  )
  ```

In this switching there is now way we can find out this switching happens because of restore wallet or loading existing wallet. i.e. if we want to do some address discover in WalletNavigator but couldn't reach  WalletProvider. 
so by creating instance there once is less expensive operation so that we are discovering the active address and setting the count of active address before initialising the wallet instance we are going to use further.
 
cc @ivan-zynesis